src/main/java/sirius/biz/util/ReportUnusedNLSKeysJob.java

```diff
@@ -0,0 +1,105 @@
+import java.util.stream.Collectors;
+
+/**
+ * Collects all unused NLS keys accross all cluster nodes and reports the intersection as text file.
+ * <p>
+ * We build the intersection here to determine the truly unused keys, as maybe some nodes have specific tasks
+ * and will not use all keys.
+ */
+@Register
+@Permission(TenantUserManager.PERMISSION_SYSTEM_ADMINISTRATOR)
+public class ReportUnusedNLSKeysJob extends SimpleBatchProcessJobFactory {
+
+    @Part
+    private InterconnectClusterManager clusterManager;
+
+    @Override
+    protected void collectParameters(Consumer<Parameter<?>> parameterCollector) {
+        // This job doesn't require any parameters.
+    }
+
+    @Override
+    protected String createProcessTitle(Map<String, String> context) {
+        return getLabel();
+    }
+
+    @Override
+    public String getLabel() {
+        return "Export all unused NLS keys";
+    }
+
+    @Nullable
+    @Override
+    public String getDescription() {
+        return "Queries all unused NLS keys on each node in the cluster, computes the intersection and reports this as textfile within the created process.";
+    }
+
+    @Override
+    protected void execute(ProcessContext process) throws Exception {
+        process.log("Fetching unused keys from all nodes...");
+        Set<String> missingKeys = new HashSet<>();
+        Monoflop mf = Monoflop.create();
+        clusterManager.callEachNode("/system/nls.unused/" + clusterManager.getClusterAPIToken())
+                      .forEach(missingKeysOfNode -> {
+                          Set<String> keysOfNode = missingKeysOfNode.getJSONArray("unused")
+                                                                    .stream()
+                                                                    .map(Object::toString)
+                                                                    .collect(Collectors.toSet());
+
+                          process.log(ProcessLog.info()
+                                                .withFormattedMessage("Received %s keys from %s",
+                                                                      keysOfNode.size(),
+                                                                      missingKeysOfNode.getString(
+                                                                              InterconnectClusterManager.RESPONSE_NODE_NAME)));
+
+                          if (mf.firstCall()) {
+                              missingKeys.addAll(keysOfNode);
+                          } else {
+                              missingKeys.retainAll(keysOfNode);
+                          }
+                      });
+
+        if (missingKeys.isEmpty()) {
+            process.log("No unused keys found...");
+        } else {
+            process.log(ProcessLog.info().withFormattedMessage("Exporting %s unused keys...", missingKeys.size()));
+            try (PrintStream out = new PrintStream(process.addFile("unused_keys.txt"))) {
```

Comment:
Why not use process.addTable? One can just visualize or export the table to a local file...

Reply:
This data is processed most probably any way (it is just a single key per line nothing more). Also, running this right after a patch might generate large datasets which will further fille the processlogs table with many more entries.